src/Tasks/Drush/Drush.php

```diff
@@ -38,6 +40,18 @@ class Drush extends Exec
      */
     public function siteInstall()
     {
+        $dbUrl = sprintf(
+            '%s://%s:%s@%s:%s/%s',
+            $this->databaseType,
+            $this->databaseUser,
+            $this->databasePassword,
+            $this->databaseHost,
+            $this->databasePort,
+            $this->databaseName
+        );
+
+        $dbUrl = (isset($this->databaseUrl) && !empty($this->databaseUrl)) ? $this->databaseUrl : $dbUrl;
```

Comment:
I would suggest to use the https://github.com/jakeasmith/http_build_url library so we can do a very simple:

```
$dbUrl = http_build_url([
  'scheme' => $this->databaseType,
  'host' => $this->databaseHost,
  'port' => $this->databasePort,
  // ...
]);
```

I had a quick look through the code and I'm not sure if they handle the case of a missing host correctly. For example in SQLite the host is empty, so the URL `sqlite://<no hostname>/path-to-db` results in a URL with three slashes `sqlite:///path-to-db`. The library doesn't have a test case for a missing host name so this might not work as expected.